src/porepy/numerics/nonlinear/nonlinear_solvers.py

```diff
@@ -121,10 +121,11 @@ def newton_step() -> None:
                 newton_step()
 
                 if is_diverged:
-                    model.after_nonlinear_failure()
```

Comment:
@keileg please take a look. `after_nonlinear_failure` is called again after `break`. This leads to calling it twice. Previously, this was not an issue since `after_nonlinear_failure` always raised an exception.

Reply:
Agree, this should be fine. But perhaps add a comment that failure will be handled below.

Reply:
I agree.